src/Sentry/buildTransitive/Sentry.targets

```diff
@@ -34,8 +34,13 @@
       <SentryUploadSymbols Condition="'$(SentryUploadSymbols)' == ''">false</SentryUploadSymbols>
       <SentryUploadSources Condition="'$(SentryUploadSources)' == ''">false</SentryUploadSources>
 
-      <!-- This property controls if the Sentry CLI is to be used at all.  Setting false will disable all Sentry CLI usage. -->
-      <UseSentryCLI Condition="'$(UseSentryCLI)' == '' And ('$(SentryUploadSymbols)' == 'true' Or '$(SentryUploadSources)' == 'true' Or $(SentryUploadAndroidProguardMapping) == 'true')">true</UseSentryCLI>
+      <!-- This property controls if the Sentry CLI is to be used at all. Setting false will disable all Sentry CLI usage.
+           We're explicitly skipping uploads for Sentry projects (except samples) because they interfere with CLI integration test asserts. -->
+      <UseSentryCLI Condition="
+        '$(UseSentryCLI)' == ''
+        and ('$(SentryUploadSymbols)' == 'true' or '$(SentryUploadSources)' == 'true' or $(SentryUploadAndroidProguardMapping) == 'true')
+        and ('$(MSBuildProjectName)' != 'Sentry')
+        and (!$(MSBuildProjectName.StartsWith('Sentry.')) or '$(MSBuildProjectName)' == '$(SentryCLIIntegrationTestProject)')">true</UseSentryCLI>
```

Comment:
`!$(MSBuildProjectName.StartsWith('Sentry.')` doesn't skip our samples?

Reply:
Yes, it does and it should. We don't want to upload samples to sentry.io just because we (sentry-dotnet SDK contributors) have SENTRY_AUTH_TOKEN env var set locally, which would enable CLI.

Instead, it will only ever upload any project starting with`Sentry.` if explicitly configured by `/p:SentryCLIIntegrationTestProject=.....`. That's what the integration test script passes now.

Reply:
Ah OK. In that case maybe [the comment](https://github.com/getsentry/sentry-dotnet/blob/1dfe043e9df4c277c3a66eb06f440d03150cbc00/src/Sentry/buildTransitive/Sentry.targets#L38-L37) needs to be updated. Currently it reads:
> We're explicitly skipping uploads for Sentry projects (except samples)

Reply:
right, that was an in-betwen solution, thanks